runtime/polkadot/src/lib.rs

```diff
@@ -1130,48 +1277,63 @@ pub type Executive = frame_executive::Executive<
 	frame_system::ChainContext<Runtime>,
 	Runtime,
 	AllPallets,
-	TechnicalMembershipStoragePrefixMigration,
+	SetInitialHostConfiguration,
 >;
 /// The payload being signed in transactions.
 pub type SignedPayload = generic::SignedPayload<Call, SignedExtra>;
 
-const TECHNICAL_MEMBERSHIP_OLD_PREFIX: &str = "Instance1Membership";
 /// Migrate from `Instance1Membership` to the new pallet prefix `TechnicalMembership`
-pub struct TechnicalMembershipStoragePrefixMigration;
+pub struct SetInitialHostConfiguration;
 
-impl OnRuntimeUpgrade for TechnicalMembershipStoragePrefixMigration {
+impl OnRuntimeUpgrade for SetInitialHostConfiguration {
 	fn on_runtime_upgrade() -> frame_support::weights::Weight {
-		use frame_support::traits::PalletInfo;
-		let name = <Runtime as frame_system::Config>::PalletInfo::name::<TechnicalMembership>()
-			.expect("TechnialMembership is part of runtime, so it has a name; qed");
-		pallet_membership::migrations::v4::migrate::<Runtime, TechnicalMembership, _>(
-			TECHNICAL_MEMBERSHIP_OLD_PREFIX,
-			name,
-		)
-	}
-
-	#[cfg(feature = "try-runtime")]
-	fn pre_upgrade() -> Result<(), &'static str> {
-		use frame_support::traits::PalletInfo;
-		let name = <Runtime as frame_system::Config>::PalletInfo::name::<TechnicalMembership>()
-			.expect("TechnicalMembership is part of runtime, so it has a name; qed");
-		pallet_membership::migrations::v4::pre_migrate::<TechnicalMembership, _>(
-			TECHNICAL_MEMBERSHIP_OLD_PREFIX,
-			name,
-		);
-		Ok(())
-	}
+		use parachains_configuration::HostConfiguration;
+
+		let active_config: HostConfiguration::<BlockNumber> = HostConfiguration {
+			max_code_size: 10_485_760,
+			max_head_data_size: 20_480,
+			max_upward_queue_count: 10,
+			max_upward_queue_size: 51_200,
+			max_upward_message_size: 51_200,
+			max_upward_message_num_per_candidate: 10,
+			hrmp_max_message_num_per_candidate: 10,
+			validation_upgrade_frequency: 14_400,
+			validation_upgrade_delay: 600,
+			max_pov_size: 5_242_880,
+			max_downward_message_size: 51_200,
+			ump_service_total_weight: 100_000_000_000,
+			hrmp_max_parachain_outbound_channels: 10,
+			hrmp_max_parathread_outbound_channels: 0,
+			hrmp_open_request_ttl: 72,
+			hrmp_sender_deposit: deposit(1004, 100 * 1024),
+			hrmp_recipient_deposit: deposit(1004, 100 * 1024),
+			hrmp_channel_max_capacity: 1_000,
+			hrmp_channel_max_total_size: 102_400,
+			hrmp_max_parachain_inbound_channels: 10,
+			hrmp_max_parathread_inbound_channels: 0,
+			hrmp_channel_max_message_size: 102_400,
+			code_retention_period: EPOCH_DURATION_IN_SLOTS * 6,
```

Comment:
What's the rationale for this number?

Reply:
AFAIK Code needs to be available for the entire dispute period, which is 6 epochs.